### PR TITLE
DCS-846 adding reasons and primary reason to report details macro

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/data/UseOfForceReport.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+export type IncidentDetails = {
```
Comment (andrewrlee) [inline server/data/UseOfForceReport.ts:1]: created a type to represent the complete form data. This is mostly right I think. 

Could look at trying to represent values which are codes of enums as a restricted set of values

File: server/services/reportSummary.ts [hunk 2]
```diff
@@ -35,8 +45,15 @@ const createIncidentDetails = (
...
-const createUseOfForceDetails = (details = {}) => {
+const createUseOfForceDetails = (
+  details: Partial<UseOfForceDetails> = {},
+  reasonsForUseOfForce: Partial<ReasonsForUseOfForce> = {}
+) => {
   return {
+    reasonsForUseOfForce: whenPresent(reasonsForUseOfForce.reasons, reasons =>
+      reasons.map(value => toLabel(UofReasons, value)).join(', ')
+    ),
+    primaryReason: whenPresent(reasonsForUseOfForce.primaryReason, value => toLabel(UofReasons, value)),
```
Comment (andrewrlee) [inline server/services/reportSummary.ts:56]: two new fields which may or may not be present when rendering reports

File: server/views/formPages/incident/select-primary-uof-reason.html [hunk 3]
```diff
@@ -13,7 +13,7 @@ <h1 class="govuk-heading-xl mainHeading">{{ pageTitle }}</h1>
     {% endfor %}
       {{ govukRadios({
         idPrefix: "primaryReason",
-        name: "primaryReason[]",
+        name: "primaryReason",
```
Comment (andrewrlee) [inline server/views/formPages/incident/select-primary-uof-reason.html:16]: this was incorrectly saving primary reasons as an array

File: server/views/pages/reportDetailMacro.njk [hunk 4]
```diff
@@ -152,6 +152,26 @@
         url:  '/report/' + bookingId + '/use-of-force-details' if bookingId
       })
     }}
+    {% if data.useOfForceDetails.reasonsForUseOfForce %}
```
Comment (andrewrlee) [inline server/views/pages/reportDetailMacro.njk:155]: only show if present

File: package.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
     "build": "npm run css-build && tsc && npm run copy-views",
     "css-build": "./bin/build-css",
     "clean": "rm -rf dist/* .port.tmp *.log build/* uploads/* test-results.xml",
-    "lint": "eslint . --cache --max-warnings 70",
+    "lint": "eslint . --cache --max-warnings 60",
```
[the remark this answers is not in the document]
Comment (GurnankCheema) [reply package.json:19]: cool

File: server/data/UseOfForceReport.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  painInducingTechniques: boolean
+  painInducingTechniquesUsed: string[]
+}
+
+export type ReasonsForUseOfForce = {
+  reasons: string[]
+  primaryReason: string
+}
+
+export type RelocationAndInjuries = {
```
Comment (GurnankCheema) [inline server/data/UseOfForceReport.ts:28]: will be adding stuff here in future commit for 'other' relocationType

File: server/services/reportSummary.ts (hunk 2 above)
Comment (GurnankCheema) [reply server/services/reportSummary.ts:56]: ok

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**